### PR TITLE
Hosted UI CLI setup for JS + Native apps

### PR DESCRIPTION
**Pull request by kaustavghosh06**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: android/authentication.md [hunk 1]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
-{: .callout .callout-info}
-
-To setup App Client:
-- Go to [Amazon Cognito Console](https://aws.amazon.com/cognito/).
-- Click *User Pools* on the top menu to select a User Pool or create a new one.
-- Click *App integration*  and *App client settings* on the left menu.
-- Select *Enabled Identity Providers* and enter *Callback URL(s)* and *Sign out URL(s)* fields.
+#### Automated Setup with CLI
+
+You would now need to configure your identity providers(Google, Facebook or Login with Amazon) which you would like to use.
```
Comment (nikhil-dabhade) [inline android/authentication.md]: You would now need to configure --> You need to configure

File: android/authentication.md [hunk 2]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+
+##### Setting Up OAuth With Facebook
+
+1. Create a [developer account with Facebook](https://developers.facebook.com/docs/facebook-login)
```
Comment (nikhil-dabhade) [inline android/authentication.md]: Add "." in the end.

File: android/authentication.md [hunk 3]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+2. [Sign In](https://developers.facebook.com/) with your Facebook credentials.
```
Comment (nikhil-dabhade) [inline android/authentication.md]: Add "." in the end.

File: android/authentication.md [hunk 4]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+![Image]({{common_media}}/cognitoHostedUI/google5.png)
+3. Create your OAuth2.0 credentials by choosing *OAuth client ID* from the *Create credentials* drop-down list.
+![Image]({{common_media}}/cognitoHostedUI/google6.png)
+4. Choose *Web application*.
+5. Click *Create* twice.
+6. Note the *OAuth client ID* and *client secret*. You will need them for the next section in the CLI flow.
+7. Choose *OK*.
+
+##### Setting up OAuth with Login with Amazon
+1. Create a [developer account with Amazon](https://developer.amazon.com/login-with-amazon).
```
Comment (nikhil-dabhade) [inline android/authentication.md]: Add "." in the end.

File: android/authentication.md [hunk 5]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+6. Note the *App ID* and the *App Secret*. You will use them in the next section in the CLI flow.
+
+##### Setting up OAuth with Google
+
+1. Go to the [Google developer console](https://console.developers.google.com).
+2. On the left navigation bar, choose *Credentials*.
+![Image]({{common_media}}/cognitoHostedUI/google5.png)
+3. Create your OAuth2.0 credentials by choosing *OAuth client ID* from the *Create credentials* drop-down list.
+![Image]({{common_media}}/cognitoHostedUI/google6.png)
+4. Choose *Web application*.
```
Comment (nikhil-dabhade) [inline android/authentication.md]: It asked me for configuring the consent screen after step 3 in Google console. Did we miss that by any chance ? We should capture that if it is the normal flow.

File: android/authentication.md [hunk 6]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+5. Click *Create* twice.
```
Comment (nikhil-dabhade) [inline android/authentication.md]: I had to click Create only once. Can you confirm ?

File: android/authentication.md [hunk 7]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+11. Type your redirect URL into *Valid OAuth Redirect URIs*. It will consist of your user pool domain with the /oauth2/idpresponse endpoint.
+
+    ```https://<your-user-pool-domain>/oauth2/idpresponse```
+
+    ![Image]({{common_media}}/cognitoHostedUI/facebook8.png)
+12. Save changes.
+
+##### Setting up Hosted UI Domain with Google
+
+1. Go to [Google Developer Console](https://developers.google.com/identity/sign-in/web/sign-in)
```
Comment (nikhil-dabhade) [inline android/authentication.md]: Add ' . ' in the end.

File: android/authentication.md [hunk 8]
```diff
@@ -1088,64 +1089,136 @@ public class AuthenticatorActivity extends Activity {
...
+![Image]({{image_base}}/cognitoHostedUI/google2.png)
+4. Type in your product name and choose *NEXT*.
+5. Choose *Web browser* from the *Where are you calling from?* drop-down list.
+![Image]({{image_base}}/cognitoHostedUI/google3.png)
+6. Click *CREATE*. You will NOT use the *Client ID* and *CLient Secret* from this step.
+7. Click Done.
+8. Go to the [Google developer console](https://console.developers.google.com).
+9. On the left navigation bar, choose *Credentials*.
+![Image]({{image_base}}/cognitoHostedUI/google5.png)
+10. Select the client you created in the first step and choose the edit option
```
Comment (nikhil-dabhade) [inline android/authentication.md]: Add ' . ' in the end.

File: ios/authentication.md [hunk 9]
```diff
@@ -872,64 +873,136 @@ Now, your drop-in UI will show a Google sign in button which the users can use t
...
-{: .callout .callout-info}
-
-To setup App Client:
-- Go to [Amazon Cognito Console](https://aws.amazon.com/cognito/).
-- Click *User Pools* on the top menu to select a User Pool or create a new one.
-- Click *App integration*  and *App client settings* on the left menu.
-- Select *Enabled Identity Providers* and enter *Callback URL(s)* and *Sign out URL(s)* fields. 
+#### Automated Setup with CLI
+
+You would now need to configure your identity providers(Google, Facebook or Login with Amazon) which you would like to use.
```
Comment (nikhil-dabhade) [inline ios/authentication.md]: You would now need to configure --> You need to configure

File: ios/authentication.md [hunk 10]
```diff
@@ -872,64 +873,136 @@ Now, your drop-in UI will show a Google sign in button which the users can use t
...
+
+##### Setting Up OAuth With Facebook
+
+1. Create a [developer account with Facebook](https://developers.facebook.com/docs/facebook-login)
```
Comment (nikhil-dabhade) [inline ios/authentication.md]: Add ' . ' in the end.

File: ios/authentication.md [hunk 11]
```diff
@@ -872,64 +873,136 @@ Now, your drop-in UI will show a Google sign in button which the users can use t
...
+6. Note the *App ID* and the *App Secret*. You will use them in the next section in the CLI flow.
+
+##### Setting up OAuth with Google
+
+1. Go to the [Google developer console](https://console.developers.google.com).
+2. On the left navigation bar, choose *Credentials*.
+![Image]({{common_media}}/cognitoHostedUI/google5.png)
+3. Create your OAuth2.0 credentials by choosing *OAuth client ID* from the *Create credentials* drop-down list.
+![Image]({{common_media}}/cognitoHostedUI/google6.png)
+4. Choose *Web application*.
```
Comment (nikhil-dabhade) [inline ios/authentication.md]: It asked me for configuring the consent screen after step 3 in Google console. Did we miss that by any chance ? We should capture that if it is the normal flow.

File: ios/authentication.md [hunk 12]
```diff
@@ -872,64 +873,136 @@ Now, your drop-in UI will show a Google sign in button which the users can use t
...
+5. Click *Create* twice.
```
Comment (nikhil-dabhade) [inline ios/authentication.md]: I did not have to click "Create" twice when trying this.

File: js/authentication.md [hunk 13]
```diff
@@ -1100,57 +1106,75 @@ Alternatively, when enabling a domain in User Pools, you can interact directly w
...
+6. Note the *App ID* and the *App Secret*. You will use them in the next section in the CLI flow.
+
+##### Setting up OAuth with Google
+
+1. Go to the [Google developer console](https://console.developers.google.com).
+2. On the left navigation bar, choose *Credentials*.
+![Image]({{common_media}}/cognitoHostedUI/google5.png)
+3. Create your OAuth2.0 credentials by choosing *OAuth client ID* from the *Create credentials* drop-down list.
+![Image]({{common_media}}/cognitoHostedUI/google6.png)
+4. Choose *Web application*.
```
Comment (nikhil-dabhade) [inline js/authentication.md]: It asked me for configuring the consent screen after step 3 in Google console. Did we miss that by any chance ? We should capture that if it is the normal flow.

File: js/authentication.md [hunk 14]
```diff
@@ -1161,29 +1185,83 @@ console.log('url', Linking.makeUrl('/'));
...
+11. Type your redirect URL into *Valid OAuth Redirect URIs*. It will consist of your user pool domain with the /oauth2/idpresponse endpoint.
+
+    ```https://<your-user-pool-domain>/oauth2/idpresponse```
+
+    ![Image]({{common_media}}/cognitoHostedUI/facebook8.png)
+12. Save changes.
+
+##### Setting up Hosted UI Domain with Google
+
+1. Go to [Google Developer Console](https://developers.google.com/identity/sign-in/web/sign-in)
```
Comment (nikhil-dabhade) [inline js/authentication.md]: Add ' . ' in the end.

File: ios/authentication.md (hunk 11 above)
Comment (kaustavghosh06) [reply ios/authentication.md]: I don't think so this is a part of the normal flow. I used this - https://aws-amplify.github.io/docs/js/cognito-hosted-ui-federated-identity as a reference for the google provider setup flow.

File: android/authentication.md (hunk 1 above)
Comment (kaustavghosh06) [reply android/authentication.md]: Check

File: android/authentication.md (hunk 6 above)
Comment (kaustavghosh06) [reply android/authentication.md]: You're right. Removed "twice".

File: ios/authentication.md (hunk 11 above)
Comment (nikhil-dabhade) [reply ios/authentication.md]: I think I may know why it is different. The link the doc in the PR is https://console.developers.google.com whereas in the link you pointed out the link is https://developers.google.com/identity/sign-in/web/sign-in

Comment (kaustavghosh06) [reply ios/authentication.md]: You need to have both of them. The second link comes in - in the section below during the final setup of OAuth with the hosted UI link which is an output of the `amplify push` command.

**Pull request merged**